.github/workflows/linux.yml

```diff
@@ -471,13 +471,9 @@ jobs:
     - name: Install pre-release pip
       run: |
         pip install -U --pre pip
-    # TODO: remove cython nightly install when cython does a release
-    - name: Install nightly Cython
+    - name: Install nightly NumPy and Cython
       run: |
-        pip install git+https://github.com/cython/cython
```

Comment:
You could also just add `NO_CYTHON_COMPILE=true` to the start of these lines to accomplish the same thing, if the only objective is to reduce CI time, as noted in the linked discussion.

I don't think the distinction matters much, depends on how closely you want to match the frequency of changes/breaks in Cython source vs. less frequently updated wheels.

Since both free-threaded CI jobs I'm aware of (the wheel one and regular one) both look fine here, it is probably just fine to merge this.

Reply:
Well, the upside of compiling Cython in-place is that issues such as https://github.com/scipy/scipy/pull/20717#issuecomment-2156637336 could be caught earlier, however, I don't know if the frequency of source changes in Cython is something that SciPy wants to keep track after 3.13 comes up and Cython wheels can be installed directly from PyPi

Reply:
> compiling Cython in-place

That issue was specific to a compiled version of Cython? I'm just saying that the new wheels you propose to use here are *not* compiled Cython IIUC, just the raw Python source (which is mostly fine), and the purpose of the PR to 

> reduce overall execution times

could be accomplished by either pulling in the non-compiled wheels or by pulling in the raw source from GitHub and specifying to `pip` not to compile it, and the only difference is the frequency of change. Maybe using the wheels is less likely to pick up noise.

Reply:
I don't think regressions are happing often. I like the idea of using Cython nightly wheels. The trade-off is that fixes in Cython master may take a little longer to materialize. But so far we haven't had any problems in the free-threaded build with Cython, so I think choosing faster install times and a bit more stability makes sense.

Reply:
> choosing faster install times

Ok, I don't think this one is true if you disable the compile on the `pip install`, but I think we agree on the rest.